backend/src/apiserver/visualization/license.sh

```diff
@@ -23,8 +23,15 @@
 
 # Get the list of python packages installed locally.
 IFS=$'\n'
-INSTALLED_PACKAGES=($(pip freeze | sed s/=.*//))
-
+INSTALLED_PACKAGES=($(if [ "$(uname -m)" = "x86_64" ]; then \
+                          pip freeze | sed s/=.*// ; \
+                      # As some of the AArch64 dependancies are installed from source,
+                      # running pip freeze will give additional package information after
+		      # an " @", we need to remove that part.
+		      # The license checking on some additional installed packages on AArch64 should also be skipped.
```

Comment:
Are you sure the license checking should be skipped for those packages? Maybe just add the licenses to the list.

Reply:
Hi @Ark-kun , thanks for your comment. Well, there are two cases for these skipped packages:
1. As some of the dependancies in the arm64 docker image are built from source, they need Cython to build. However I checked the x86_64 image, Cython is not installed in that image. Therefore Cython is skipped.

2. The arm64 base image proposed in this PR does not contain some of the required packages, and I have to install them manually by pip from [Dockerfile.visualization#L43](https://github.com/kubeflow/pipelines/pull/3839/files#diff-e3fa8e7715928ccd1476debc9822407bR43). This will lead to some additional packages `cffi`, `jeepney`, `pycairo`, `pycparser` also installed. As x86_64 image also does not have them, they should be skipped.

I think if I add these packages to the license list, the x86_64 building will be broken. So I chose to only skip them for arm64.

Reply:
>I think if I add these packages to the license list, the x86_64 building will be broken.

That should not be the case. The license script just uses the list to get URLs for the package licenses and downloads licenses for all packages returned by pip freeze.

Reply:
Hi @Ark-kun, thanks for your patience. 

Yes and no. The `license.sh` does work as you described. However these 4 packages are only installed in the arm64 image. If their licenses are added to the .csv file. The `license.sh` will also check the license of these 4 packages on x86_64, and will ask us to install them in x86_64 images, which is unnecessary and break the `docker build` on x86_64 as shown in [license.sh#L84](https://github.com/kubeflow/pipelines/blob/ef73aab59f8fc06db19d5e6f0ca35337e98bcf77/backend/src/apiserver/visualization/license.sh#L84).

Reply:
Oh. The license.sh we use for components does not have that part. https://github.com/kubeflow/pipelines/blob/ef73aab59f8fc06db19d5e6f0ca35337e98bcf77/components/license.sh

I think we should remove `exit 1` from that script. You can do that.

Reply:
Hi @Ark-kun! Thanks, yes that is a solution, however I am worried a little bit about this removing the `exit 1`....as I personally think it is a little bit too aggressive. Personally I would like to achieve the goal that we can completely encapsulate the arm64 stuff without affecting the x86_64.

Also I did a quick test about adding the licenses in the .csv file, and it seems if I add them to the csv, I have to run the `license-download.sh` and also commit the newly downloaded licenses in this PR....Would that be also ok?

Thanks again for your patience.

Reply:
>I am worried a little bit about this removing the exit 1....as I personally think it is a little bit too aggressive.

I think that the script should fail when there is a missing license. But an extra license is fine in my opinion.

>Also I did a quick test about adding the licenses in the .csv file, and it seems if I add them to the csv, I have to run the license-download.sh and also commit these newly downloaded licenses in this PR....Would that be also ok?

I think this would be OK.

I consider it more readable and safe than hacking the license.sh script.

Reply:
Thanks @Ark-kun. That would be awesome. I have updated related files in this PR and added everything you suggested. Would you mind taking a look again? Thanks!

Reply:
It will be harder to remove obsolete license files, because we don't know which packages are not needed in x86 but needed in arm, even though the message exists.
Obsolete licenses is not a big issue I agree, just not ideal.